### ISSUE TITLE
Increase/Decrease Font

### ISSUE THREAD

**Issue by thequbit** [gitterHQ/desktop#28]
It would be great to be able to increase/decrease font size in the chat window for use on smaller screens, as well as display boards.  After talking with support this seems possible, however not yet implemented.

Ctrl +, or a +/- button would be great.

Thank you for the continued development of gitter!

**Comment by thequbit** [comment]
@trevorah is someone working on this? I'm not super familiar with the technologies used, but I can take a stab at it if you are accepting PRs.

**Comment by trevorah** [comment]
No-ones working on this right now, but you would be my favourite person if you sent a pull request.

**Comment by thequbit** [comment]
Looks like registering a gloab hot key shortcut would be a great solution here:

https://github.com/nwjs/nw.js/wiki/Shortcut

That would call the zoom level ( as @suprememoocow suggested ).

https://github.com/nwjs/nw.js/wiki/Window#windowzoomlevel

Looks like there is *quite* the stack to setup to get this thing compiled ... I'll fiddle and try and get everything compiled this week.

Assigning to myself.

**Comment by trevorah** [comment]
Cool beans. If you have any issues with getting things running or existing code that doesnt make sense, please let me know. Its probably something that we've messed up that needs fixing.

**Comment by seancorfield** [comment]
Has there been any progress on this? The default font size on Windows is ridiculously small. Is there any way to change that default?

**Comment by MadLittleMods** [comment]
@seancorfield No progress, you can open up the dev tools and adjust the size. Or adjust the CSS and make a custom build from this repo.

**Comment by JernejJerin** [comment]
:+1:, I am also missing this setting due to this [problem](https://github.com/gitterHQ/gitter/issues/878).

**Comment by choppsv1** [comment]
+1, also @MadLittleMods I've built the desktop app locally but I can't seem to override the values the app loads with router-chat.css (can't even find that file) for changing the font size. Can you be more specific on how to change the values in css?

**Comment by MadLittleMods** [comment]
@choppsv1 Use some JS to inject some CSS into the iframes. You probably won't be able to use the nice multiline template strings in the code snippet below until I [merge and release v3.0.0](https://github.com/gitterHQ/desktop/pull/100)(which will be today) but you can always use normal string concatenation or read from a file.

```js
var gitterRootDoc = document.querySelector('#mainframe').contentWindow.document;
var gitterChatAreaDoc = gitterRootDoc.querySelector('#content-frame').contentWindow.document;


var newStyleElement1 = doc.createElement('style');
newStyleElement1.innerHTML = `
* {
	background-color: #f00;
}
`;
gitterRootDoc.body.appendChild(newStyleElement1);


var newStyleElement2 = doc.createElement('style');
newStyleElement2.innerHTML = `
* {
	background-color: #0f0;
}
`;
gitterChatAreaDoc.body.appendChild(newStyleElement2);
```

**Comment by choppsv1** [comment]
So when I set the background color for the chat area to #222 and color to #fff all the chat entries are invisible, the chat area is just filled with the background. This is the case even if I set the chat areas to other colors; however, if I don't set the background and do set the chat color that does work. Not sure why setting the background is masking the more specific chat class settings. Here's the 2 examples:
All Background Color Only:
```js
win.zoomLevel = 3;

// Wait for the iframes to load in
setTimeout(function() {
    var gitterRootDoc = win.window.document.querySelector('#mainframe').contentWindow.document;
    var gitterChatAreaDoc = gitterRootDoc.querySelector('#content-frame').contentWindow.document;

    var newStyleElement1 = gitterRootDoc.createElement('style');
    newStyleElement1.innerHTML = " * { background-color: #222; color: #fff; } ";
    gitterRootDoc.body.appendChild(newStyleElement1);

    var newStyleElement2 = gitterRootDoc.createElement('style');
    newStyleElement2.innerHTML = " * { background-color: #222; color: #fff;} ";
    gitterChatAreaDoc.body.appendChild(newStyleElement2);

    var newStyleElement3 = gitterChatAreaDoc.createElement('style');
    newStyleElement3.innerHTML = " .chat-item { color: #eee; background-color: #222; } ";
    gitterChatAreaDoc.body.appendChild(newStyleElement3);


}, 4000);
```

Chat's black background white foreground:

```js
win.zoomLevel = 3;

// Wait for the iframes to load in
setTimeout(function() {
    var gitterRootDoc = win.window.document.querySelector('#mainframe').contentWindow.document;
    var gitterChatAreaDoc = gitterRootDoc.querySelector('#content-frame').contentWindow.document;

    var newStyleElement1 = gitterRootDoc.createElement('style');
    newStyleElement1.innerHTML = " * { background-color: #222; color: #fff; } ";
    gitterRootDoc.body.appendChild(newStyleElement1);

    var newStyleElement3 = gitterChatAreaDoc.createElement('style');
    newStyleElement3.innerHTML = " .chat-item { color: #eee; background-color: #222; } ";
    gitterChatAreaDoc.body.appendChild(newStyleElement3);
}, 4000);
```

**Comment by MadLittleMods** [comment]
@choppsv1 This seems like a CSS issue that you will need to figure out somewhere else and not clutter up the issue. I provided a solution but you will need to discuss elsewhere like https://gitter.im/gitterHQ/developers

**Comment by choppsv1** [comment]
I had deleted the comment within minutes of posting.. Not sure if you noticed.

**Comment by kulicuu** [comment]
+1

**Comment by raineorshine** [comment]
When I run `var win = gui.Window.get();` in the dev tools console I get `ReferenceError: Can't find variable: gui`.

**Comment by alfiepates** [comment]
So, I was pointed to this issue from issue #1663 which I opened. 

It's kinda disappointing to see no progress has been made on this in a year for what is honestly a pretty important accessibility feature for some people. 

Gitter is `nw.js`-based, right? `nw.js` presents a pretty damn good API for this stuff: The Gitter UI literally only needs to set `Window.zoomLevel` plus or minus some value (where `0` is default).
